modules/globebrowsing/util/globelabelscomponent.cpp

```diff
@@ -0,0 +1,846 @@
+        {
+            LabelsFadeOutEnabledInfo.identifier,
+            new BoolVerifier,
+            Optional::Yes,
+            LabelsFadeOutEnabledInfo.description
+        },
+        {
+            LabelsDisableCullingEnabledInfo.identifier,
+            new BoolVerifier,
+            Optional::Yes,
+            LabelsDisableCullingEnabledInfo.description
+        },
+        {
+            LabelsDistanceEPSInfo.identifier,
+            new DoubleVerifier,
+            Optional::Yes,
+            LabelsDistanceEPSInfo.description
+        },
+        {
+            LabelAlignmentOptionInfo.identifier,
+            new StringVerifier,
+            Optional::Yes,
+            LabelAlignmentOptionInfo.description
+        },
+    }
+    };
+}
+
+GlobeLabelsComponent::GlobeLabelsComponent()
+    : properties::PropertyOwner({ "Labels" })
+    , _labelsEnabled(LabelsInfo, false)
+    , _labelsFontSize(LabelsFontSizeInfo, 30, 1, 120)
+    , _labelsMaxSize(LabelsMaxSizeInfo, 300, 10, 1000)
+    , _labelsMinSize(LabelsMinSizeInfo, 4, 1, 100)
+    , _labelsSize(LabelsSizeInfo, 2.5, 0, 30)
+    , _labelsMinHeight(LabelsMinHeightInfo, 100.0, 0.0, 10000.0)
+    , _labelsColor(LabelsColorInfo, glm::vec4(1.f, 1.f, 0.f, 1.f),
+        glm::vec4(0.f), glm::vec4(1.f))
+    , _labelsFadeInDist(LabelsFadeInStartingDistanceInfo, 1E6, 1E3, 1E8)
+    , _labelsFadeOutDist(LabelsFadeOutStartingDistanceInfo, 1E4, 1, 1E7)
+    , _labelsFadeInEnabled(LabelsFadeInEnabledInfo, true)
+    , _labelsFadeOutEnabled(LabelsFadeOutEnabledInfo, true)
+    , _labelsDisableCullingEnabled(LabelsDisableCullingEnabledInfo, false)
+    , _labelsDistaneEPS(LabelsDistanceEPSInfo, 100000.f, 1000.f, 10000000.f)
+    , _labelAlignmentOption(LabelAlignmentOptionInfo, properties::OptionProperty::DisplayType::Dropdown)
+{
+    addProperty(_labelsEnabled);
+    addProperty(_labelsFontSize);
+    addProperty(_labelsSize);
+    addProperty(_labelsMinHeight);
+    _labelsColor.setViewOption(properties::Property::ViewOptions::Color);
+    addProperty(_labelsColor);
+    addProperty(_labelsFadeInDist);
+    addProperty(_labelsFadeOutDist);
+    addProperty(_labelsMinSize);
+    addProperty(_labelsFadeInEnabled);
+    addProperty(_labelsFadeOutEnabled);
+    addProperty(_labelsDisableCullingEnabled);
+    addProperty(_labelsDistaneEPS);
+
+    _labelAlignmentOption.addOption(0, "Horizontally");
+    _labelAlignmentOption.addOption(1, "Circularly");
+    _labelAlignmentOption = Horizontally;
+    addProperty(_labelAlignmentOption);
+}
+
+void GlobeLabelsComponent::initialize(const ghoul::Dictionary& dictionary, 
+    globebrowsing::RenderableGlobe* globe, std::shared_ptr<ghoul::fontrendering::Font> font)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "GlobeLabelsComponent"
+    );
+        
+    _globe = globe;
+
+    // Reads labels' file and build cache file if necessary
+    _labelsDataPresent = false;
+    if (!dictionary.empty()) {
```

Comment:
Code quality thing.  Instead of nesting this many `if` statements, i'd be better to early out when possible:

```
if (dictionary.empty()) {
  return;
}
...
if (!successLabels) {
  return;
}
```

That will reduce the overall amount of whitespaces